RiotSwiftUI/Modules/Room/PollHistory/PollHistoryDetail/PollHistoryDetailModels.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+// MARK: - Coordinator
+
+typealias PollHistoryDetailViewModelCallback = (PollHistoryDetailViewModelResult) -> Void
+
+enum PollHistoryDetailViewModelResult {
+    case dismiss
+    case viewInTimeline
+}
+
+// MARK: View model
+
+
+
+// MARK: View
+
+struct PollHistoryDetailViewState: BindableState {
+    var timelineViewModel: TimelinePollViewModelProtocol
```

Comment:
I think you just need a `TimelinePollViewModelType.Context` here instead of the full view model right?

Reply:
Sorry for jumping in here but neither the viewModel nor the context from a different view hierarchy have any business belonging here. If passing data between 2 view hierarchies is what you're looking for you should map it between their own data types. If you need them in sync you can use publishers. The whole point to these templates is to keep things neatly separated.

Reply:
The problem @flescio is trying to solve here is this:

`PollHistoryDetail` embeds a full `TimelinePollView` which needs its context.

How do you suggest to do that?

Reply:
@stefanceriu The point here was to try to fully embed existing logic and view. I'm not sure which pattern to follow. but in order to embed the TimelinePollView, we need the Context of its viewModel.